coopdevs/timeoverflow
# Statistics code clean up

This code needs further refactoring (probably move all logic to Services/Decorators), this is just a 1st minor step cleaning some details here and there and adding some tests.

**SUMMARY OF CHANGES**
- remove redundant and too much verbose `statistics_` prefix in files, methods, keys, routes, ...
- remove unnecessary and confusing @ivars
- remove useless comments
- rename some vars (ini/fi -> from/to)
- some code re-arrange and re-grouping (**NO logic changes**) 
- improved test coverage ⬆️ 

**NOTES** 
- ⚠️  Some translations keys need to be **renamed** in `Localeapp` after merging this (https://github.com/coopdevs/timeoverflow/pull/600/files#diff-44438ce218f5287c58d0017f965d888715635d94280669896f75841fbd7b4cd7) -> basically remove the `statistics_`prefix under the `statistics.*` scope. Also remove a couple of deprecated keys. @sseerrggii not sure if the rename can be done via the UI, alternatively we can check the CLI provided by the gem.

**EXTRAS**
In the Admin section, add a "Recent Posts" box, inline with "Recent users" and "Recent Orgs" (489dab611d9990b0950738048822f8adadd080b5)

**REVIEW LINK (with no whitespace changes)** 
https://github.com/coopdevs/timeoverflow/pull/600/files?diff=unified&w=1

**QA**
Business logic was **not changed** at all, so a quick visit on each stats page to see everything keeps displaying data is enough :)
